mailserver/registry/verifier.go

```diff
@@ -0,0 +1,34 @@
+package registry
+
+import (
+	"context"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/p2p/discover"
+)
+
+type RegistryVerifier struct {
+	rc         *RegistryCaller
+	pendingOpt bool
+}
+
+func NewVerifier(contractCaller bind.ContractCaller, contractAddress common.Address) (*RegistryVerifier, error) {
+	rc, err := NewRegistryCaller(contractAddress, contractCaller)
+	if err != nil {
+		return nil, err
+	}
+
+	return &RegistryVerifier{
+		rc: rc,
+	}, nil
+}
+
+func (v *RegistryVerifier) VerifyNode(_ context.Context, nodeID discover.NodeID) bool {
+	res, err := v.rc.Exists(&bind.CallOpts{Pending: v.pendingOpt}, nodeID.Bytes())
```

Comment:
why registry doesn't use address?

Reply:
@dshulyak the address is only passed to the `RegistryCaller` which internally creates a `BoundContract` that contains that address

Reply:
no, this is not what i meant
by design we agreed that registry will hold addresses not the public key (node id bytes)

Reply:
from https://github.com/status-im/contracts/blob/280-disc-mail-servers/contracts/registries/AddressRegistry.sol#L21 
 
or any other registry from that file uses addresses, so it is better to be specific about it, otherwise there will be a problem when someone will register an address and we will try to find a node id instead